app-arch/brotli/brotli-1.0.2.recipe

```diff
@@ -0,0 +1,143 @@
+SUMMARY="Generic-purpose lossless compression algorithm"
+DESCRIPTION="A generic-purpose lossless compression algorithm that \
+compresses data using a combination of a modern variant of the LZ77 algorithm, \
+Huffman coding and 2nd order context modeling, with a compression ratio \
+comparable to the best currently available general-purpose compression methods. \
+It is similar in speed with deflate but offers more dense compression"
+HOMEPAGE="https://github.com/google/brotli"
+COPYRIGHT="2009, 2010, 2013-2016 Brotli Authors"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/google/brotli/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="c2cf2a16646b44771a4109bb21218c8e2d952babb827796eb8a800c1f94b7422"
+
+ARCHITECTURES="x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+# For now the library will follow $portVersion,
+# but upstream said that will change in the future.
+libVersion="1.0.2"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	brotli$secondaryArchSuffix = $portVersion
+	cmd:brotli$secondaryArchSuffix = $portVersion
+	lib:libbrotlicommon$secondaryArchSuffix = $libVersionCompat
+	lib:libbrotlidec$secondaryArchSuffix = $libVersionCompat
+	lib:libbrotlienc$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	brotli${secondaryArchSuffix}_devel = $portVersion
+	devel:libbrotlicommon$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlicommon_static$secondaryArchSuffix = $libVersion
+	devel:libbrotlidec$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlidec_static$secondaryArchSuffix = $libVersion
+	devel:libbrotlienc$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlienc_static$secondaryArchSuffix = $libVersion
+	"
+REQUIRES_devel="
+	brotli$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+PYTHON_PACKAGES=(python python3)
+PYTHON_VERSIONS=(2.7 3.6)
+# We don't have python2 for secondaryArch,
+# and gcc2 doesn't support the flags passed by python3
+if [[ "$effectiveTargetArchitecture" != "$targetArchitecture" ]]; then
+	unset PYTHON_PACKAGES[0]
+	unset PYTHON_VERSIONS[0]
+elif [[ "$effectiveTargetArchitecture" == "x86_gcc2" ]]; then
+	unset PYTHON_PACKAGES[1]
+	unset PYTHON_VERSIONS[1]
+fi
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	${portName}_$pythonPackage = $portVersion\
+	\"; \
+REQUIRES_$pythonPackage=\"\
+	haiku\n\
+	cmd:python$pythonVersion\
+	\""
+BUILD_REQUIRES="$BUILD_REQUIRES
+	setuptools_$pythonPackage"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:python$pythonVersion"
+done
+
+if [[ "$effectiveTargetArchitecture" == "$targetArchitecture" ]]; then
```

Comment:
It would be better to use sh-only syntax (instead of ksh/bash), at least for where sh is enough.
```diff
-if [[ "$effectiveTargetArchitecture" == "$targetArchitecture" ]]; then
+if [ "$effectiveTargetArchitecture" = "$targetArchitecture" ]; then
```
Remark: the test above could also be written this way:
```
if [ -z "$secondaryArchSuffix" ]; then
```
This is because secondaryArchSuffix is only defined when we are building for a secondary arch.
That said, your test is more readable. It is also more flexible. Image someone becomes crazy and decides to build a custom hybrid Haiku with a secondary arch identical to the primary arch. Your test continues to produce correct packages, even with weird configurations ;-)

Same suggestion 8 lines below:
```diff
-if [[ "$effectiveTargetArchitecture" != "x86_gcc2" ]]; then
+if [ "$effectiveTargetArchitecture" != "x86_gcc2" ]; then
```

I'm quite sure some parts, later, are not compatible with sh, but having the changes above will reduce the "risk" of other editors switching to ksh/bash syntax (after reading this recipe) in other recipes where sh is sufficient.